src/date-input/date-input.tsx

```diff
@@ -49,6 +54,8 @@ export const DateInput = ({
 
     const nodeRef = useRef<HTMLDivElement>(null);
     const inputRef = useRef<StandaloneDateInputRef>(null);
+    const calendarRef = useRef<InternalCalendarRef>(null);
+    const blurFired = useRef<boolean>(false); // To guard against multiple blur events from handleClose and handleBlur
```

Comment:
checking if date input is also able to avoid using the blurFired flag now too based on the changes for date range input